backend/dal/Helpers/Extensions/EntityEntryExtensions.cs

```diff
@@ -17,23 +17,42 @@ public static class EntityEntryExtensions
         /// <param name="entry"></param>
         public static void UndoAppCreatedEdits(this EntityEntry entry)
         {
-            if (entry.Entity is BaseAppEntity entity)
+            if (entry.Entity is IDisableBaseAppEntity entity)
             {
-                var oCreatedOn = (DateTime)entry.OriginalValues[nameof(entity.CreatedOn)];
-                if (!oCreatedOn.Equals(entity.CreatedOn))
-                    entity.CreatedOn = oCreatedOn;
+                var oCreatedOn = (DateTime)entry.OriginalValues[nameof(entity.AppCreateTimestamp)];
+                if (!oCreatedOn.Equals(entity.AppCreateTimestamp))
+                    entity.AppCreateTimestamp = oCreatedOn;
 
-                var oCreatedBy = (string)entry.OriginalValues[nameof(entity.CreatedBy)];
-                if (String.IsNullOrWhiteSpace(entity.CreatedBy) || !oCreatedBy.Equals(entity.CreatedBy))
-                    entity.CreatedBy = oCreatedBy;
+                var oCreatedBy = (string)entry.OriginalValues[nameof(entity.AppCreateUserid)];
+                if (String.IsNullOrWhiteSpace(entity.AppCreateUserid) || !oCreatedBy.Equals(entity.AppCreateUserid))
+                    entity.AppCreateUserid = oCreatedBy;
 
-                var oCreatedByKey = (Guid?)entry.OriginalValues[nameof(entity.CreatedByKey)];
-                if (!entity.CreatedByKey.HasValue || !oCreatedByKey.Equals(entity.CreatedByKey))
-                    entity.CreatedByKey = oCreatedByKey;
+                var oCreatedByKey = (Guid?)entry.OriginalValues[nameof(entity.AppCreateUserGuid)];
+                if (!entity.AppCreateUserGuid.HasValue || !oCreatedByKey.Equals(entity.AppCreateUserGuid))
+                    entity.AppCreateUserGuid = oCreatedByKey;
 
-                var oCreatedByDirectory = (string)entry.OriginalValues[nameof(entity.CreatedByDirectory)];
-                if (String.IsNullOrWhiteSpace(entity.CreatedByDirectory) || !oCreatedByDirectory.Equals(entity.CreatedByDirectory))
-                    entity.CreatedByDirectory = oCreatedByDirectory;
+                var oCreatedByDirectory = (string)entry.OriginalValues[nameof(entity.AppCreateUserDirectory)];
+                if (String.IsNullOrWhiteSpace(entity.AppCreateUserDirectory) || !oCreatedByDirectory.Equals(entity.AppCreateUserDirectory))
+                    entity.AppCreateUserDirectory = oCreatedByDirectory;
+            }
+        }
+
+        /// <summary>
+        /// Do not allow editing the App created properties.
+        /// Reset them to their original values.
+        /// </summary>
+        /// <param name="entry"></param>
+        public static void UndoDbCreatedEdits(this EntityEntry entry)
```

Comment:
we also track db* properties now (code first ignored these) since we are tracking them we need to make sure we don't modify them.